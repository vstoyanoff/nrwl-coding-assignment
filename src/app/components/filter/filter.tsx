import React from 'react';
import {
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  FormHelperText,
} from '@material-ui/core';

export type FilterValue = '' | 'completed' | 'not-completed';

interface FilterProps {
  filterValue: FilterValue;
  handleFilter: React.Dispatch<React.SetStateAction<FilterValue>>;
}

export function Filter({
  filterValue = '',
  handleFilter,
}: FilterProps): React.ReactElement {
  return (
    <FormControl>
      <InputLabel>Status</InputLabel>
      <Select
        value={filterValue}
        onChange={(event: React.ChangeEvent<{ value: unknown }>) =>
          handleFilter(event.target.value as React.SetStateAction<FilterValue>)
        }>
        <MenuItem value=''>
          <em>All</em>
        </MenuItem>
        <MenuItem value='completed'>Completed</MenuItem>
        <MenuItem value='not-completed'>Not completed</MenuItem>
      </Select>
      <FormHelperText>Filter tickets by status</FormHelperText>
    </FormControl>
  );
}
