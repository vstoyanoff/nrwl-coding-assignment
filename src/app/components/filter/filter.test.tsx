import React from 'react';
import { render } from '@testing-library/react';

import { Filter } from './filter';

describe('Filter bar', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Filter filterValue='' handleFilter={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
