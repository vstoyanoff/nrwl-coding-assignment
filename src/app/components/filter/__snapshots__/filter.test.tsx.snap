// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter bar renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
      Status
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        aria-haspopup="listbox"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        role="button"
        tabindex="0"
      >
        <span>
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root"
    >
      Filter tickets by status
    </p>
  </div>
</div>
`;
